src/components/Editor/DebugLine.js

```diff
@@ -46,32 +53,44 @@ export class DebugLine extends Component<Props, State> {
   }
 
   setDebugLine(
+    pauseInfo: Object,
     selectedFrame: Object,
     selectedLocation: Object,
     editor: Object
   ) {
     if (!selectedFrame) {
       return;
     }
-    const { location, location: { sourceId } } = selectedFrame;
-    const { line, column } = toEditorPosition(sourceId, location);
 
+    const { location, location: { sourceId } } = selectedFrame;
     const doc = getDocument(sourceId);
     if (!doc) {
       return;
     }
 
-    doc.addLineClass(line, "line", "new-debug-line");
+    const { line, column } = toEditorPosition(sourceId, location);
+
     // make sure the line is visible
     if (editor && editor.alignLine) {
       editor.alignLine(line);
     }
 
-    const debugExpression = markText(editor, "debug-expression", {
-      start: { line, column },
-      end: { line, column: null }
-    });
-    this.setState({ debugExpression });
+    if (pauseInfo && pauseInfo.why.type === "exception") {
+      doc.addLineClass(line, "line", "new-debug-line-error");
+      const debugExpression = markText(editor, "debug-expression-error", {
+        start: { line, column },
+        end: { line, column: null }
+      });
+      this.setState({ debugExpression });
+    } else {
+      doc.addLineClass(line, "line", "new-debug-line");
```

Comment:
these two branches are almost exactly the same, with the exception of the the class names. how about something like this:

```javascript
type TextClasses = {
  markTextClass: string,
  lineClass: string
}
getTextClasses(pauseInfo: Object): TextClasses {
  if (pauseInfo && pauseInfo.why.type === "exception") {
    return { markTextClass: "debug-expression-error", lineClass: "debug-expression-error" }
  }
  return { markTextClass: "debug-expression", lineClass: "new-debug-line" }
}

// ...

const { markTextClass, lineClass } = getTextClasses(pauseInfo);
doc.addLineClass(line, "line", markTextClass);
const debugExpression = markText(editor, lineClass, {
  start: { line, column },
  end: { line, column: null }
});
this.setState({ debugExpression });
```

this way, it is clear what is changing in the code, its not the entire if block, but just the class names

Reply:
You are right. Will change that